mk/programs.mk

```diff
@@ -32,7 +32,7 @@ define build-program
   $$(eval $$(call create-dir, $$(_d)))
 
   $$($(1)_PATH): $$($(1)_OBJS) $$(_libs) | $$(_d)/
-	$$(trace-ld) $(CXX) -o $$@ $$(LDFLAGS) $$(GLOBAL_LDFLAGS) $$($(1)_OBJS) $$($(1)_LDFLAGS) $$(foreach lib, $$($(1)_LIBS), $$($$(lib)_LDFLAGS_USE))
+	$$(trace-ld) $(CXX) -o $$@ $$(LDFLAGS) $$(GLOBAL_LDFLAGS) $$($(1)_OBJS) $$($(1)_LDFLAGS) $$(foreach lib, $$($(1)_LIBS), $$($$(lib)_LDFLAGS_USE)) -lz
```

Comment:
this probably shouldn't be in upstream

Reply:
maybe make it conditional on static? by setting GLOBAL_LDFLAGS_AFTER

Reply:
Well I think it's harmless in non-static ones? All this blah blah_after stuff is still hacky.

Reply:
Yeah i guess. Maybe just needs a big fixme or something. Libz is in everything so maybe no one will mind.

Reply:
Maybe we can put this in `Makefile` or `local.mk` at toplevel (e.g. `GLOBAL_LDFLAGS = -lz`). But it would be better to fix whatever package needs libz implicitly :-)

Reply:
https://github.com/libarchive/libarchive/issues/1446 I finally found the underlying issue! Will make a workaround in `configure.ac`.